### PR TITLE
JESI-2566

### PR DESCRIPTION
**Pull request by AndreTheHunter**
[JESI-2566](https://jesims.atlassian.net/browse/JESI-2566)
And fixes https://github.com/metabase/toucan/issues/3
- [x] Fix commit history by squashing
- [x] Create PR back into metabase/toucan

### CODE REVIEW

File: src/toucan/db.clj [hunk 1]
```diff
@@ -746,14 +747,11 @@
...
+     (delete! Database :id 1)"
   {:style/indent 1}
   [model & conditions]
-  (let [model (resolve-model model)]
+  (let [model       (resolve-model model)
+        primary-key (models/primary-key model)]
     (doseq [object (apply select model conditions)]
       (models/pre-delete object)
-      (simple-delete! model :id (:id object)))))
+      (simple-delete! model primary-key (primary-key object)))))
```
Comment (axrs) [inline src/toucan/db.clj:755]: Is there any chance the primary key is not a keyword and possibly a string? If there is such a scenario, then `("primary-key" object)` will break.

Comment (axrs) [reply src/toucan/db.clj:755]: NVM... Question answered in function return type

**Review by axrs — APPROVED** [review]

File: src/toucan/db.clj (hunk 1 above)
Comment (AndreTheHunter) [reply src/toucan/db.clj:755]: Currently we'll keep it simple, and support keywords. In the future we'll support composite keys, so we can look at string keys then. We'll see what Cam says

**Review by axrs — APPROVED** [review]
👍

**Pull request merged**